cpp/open3d/t/geometry/kernel/Metrics.cpp

```diff
@@ -0,0 +1,58 @@
+// ----------------------------------------------------------------------------
+// -                        Open3D: www.open3d.org                            -
+// ----------------------------------------------------------------------------
+// Copyright (c) 2018-2024 www.open3d.org
+// SPDX-License-Identifier: MIT
+// ----------------------------------------------------------------------------
+
+#include <vector>
+
+#include "open3d/core/Tensor.h"
+#include "open3d/t/geometry/Geometry.h"
+namespace open3d {
+namespace t {
+namespace geometry {
+
+std::vector<float> ComputeDistanceCommon(core::Tensor distance12,
+                                         core::Tensor distance21,
+                                         std::vector<Metric> metrics,
+                                         MetricParameters params) {
+    std::vector<float> metric_values;
+    float metric_val;
+
+    for (Metric metric : metrics) {
+        switch (metric) {
+            case Metric::ChamferDistance:
+                metric_val = 0.5 *
```

Comment:
I am still checking this but it looks like we should drop the 0.5 to be compatible with other libs

Reply:
https://fwilliams.info/point-cloud-utils/sections/shape_metrics/
This uses 0.5.
Kaolin is customizable, but default is 1.
https://kaolin.readthedocs.io/en/latest/modules/kaolin.metrics.pointcloud.html

I think PDAL is a good standard to follow here, and that doesn't have the 0.5, so I took it off. Let me know what you think:

https://pdal.io/en/2.4.3/apps/chamfer.html

Also PDAL has Hausodorff, which seems common in point cloud analysis (perhaps not for ML). Added that as well.